component/registry.go

```diff
@@ -44,7 +44,7 @@ type Module interface {
 	// LoadConfig parses River config and loads it into the Module.
 	// LoadConfig can be called multiple times, and called prior to
 	// [Module.Run].
-	LoadConfig(config []byte, args map[string]any) error
+	LoadConfig(config []byte, args map[string]any, moduleDefinitions map[string]string) error
```

Comment:
I introduced a new type LoaderConfigOptions that can be used when loading a new config to pass specific options. In this case it is used to pass the additional declare contents but in the future it can be extended if needed.